Lib/lib2to3/Grammar.txt

```diff
@@ -161,9 +161,26 @@ argument: ( test [comp_for] |
 	    star_expr )
 
 comp_iter: comp_for | comp_if
-comp_for: [ASYNC] 'for' exprlist 'in' testlist_safe [comp_iter]
+comp_for: [ASYNC] 'for' exprlist 'in' or_test [comp_iter]
 comp_if: 'if' old_test [comp_iter]
 
+# See note above regarding testlist_safe to see why testlist_safe is
+# necessary. Unfortunately we can't use it indiscriminately in all derivations
+# using comp_for-like pattern because testlist_safe derivation contains
+# comma which clashes with trailing commas in some derivations that used to
+# indirectly use testlist_safe (for example arglist -> argument -> test [comp_for]).
+#
+# The above created an ambiguity in the syntax. Since it seems that testlist_safe
+# was created specifically to handle list comprehensions and generator expressions
+# enclosed with parentheses I believe it's safe to only use it in those. That
+# avoids the ambiguity mentioned above.
+#
```

Comment:
I agree with the need for a verbose comment here, since it's not immediately obvious why `old_comp_iter`, etc. are necessary.  IMHO, you should remove overly cautious words like "seems" and "I believe".  Also, I think it might be worth adding a reference to this PR and/or bpo-27494 in the comment itself.